topic_dynamics/parsing.py

```diff
@@ -223,56 +250,46 @@ def calculate_diffs(slices_tokens_dir: str, output_dir: str, name: str, dates: L
+                    old_address = token_line.address.replace(dates[date - 1].strftime("%Y-%m-%d"),
+                                                             dates[date - 2].strftime("%Y-%m-%d"), 1)
                     # Check if the file with this name existed in the previous version
                     if old_address in previous_version.keys():
-                        old_tokens = {}
-                        for token in previous_version[old_address].split(","):
-                            old_tokens[token.split(":")[0]] = int(token.split(":")[1])
-                        old_tokens = Counter(old_tokens)
-                        # Calcualate which tokens have been added and removed between versions
+                        old_tokens = previous_version[old_address]
+                        # Calculate which tokens have been added and removed between versions
                         created_tokens = sorted((tokens - old_tokens).items(), key=itemgetter(1), reverse=True)
                         deleted_tokens = sorted((old_tokens - tokens).items(), key=itemgetter(1), reverse=True)
                         new_tokens = []
                         if len(created_tokens) != 0:
-                            for token in created_tokens:
-                                new_token = "+" + token[0]
-                                new_tokens.append([new_token, token[1]])
+                            new_tokens = differentiate_tokens(created_tokens, "+", new_tokens)
                         if len(deleted_tokens) != 0:
-                            for token in deleted_tokens:
-                                new_token = "-" + token[0]
-                                new_tokens.append([new_token, token[1]])
+                            new_tokens = differentiate_tokens(deleted_tokens, "-", new_tokens)
                     # If the file is new, all of its tokens are considered created
                     else:
                         tokens = sorted(tokens.items(), key=itemgetter(1), reverse=True)
                         new_tokens = []
-                        for token in tokens:
-                            new_token = "+" + token[0]
-                            new_tokens.append([new_token, token[1]])
+                        new_tokens = differentiate_tokens(tokens, "+", new_tokens)
                     if len(new_tokens) != 0:
                         formatted_new_tokens = transform_identifiers(new_tokens)
                         count_index_diff = count_index_diff + 1
-                        fout.write(str(count_index_diff) + ";" + address + ";" + ",".join(formatted_new_tokens) + "\n")
+                        fout.write(str(count_index_diff) + ";" + token_line.address + ";"
+                                   + ",".join(formatted_new_tokens) + "\n")
             # Iterate over files in the "previous" version to see which have been deleted
             for address in previous_version.keys():
```

Comment:
what is address?

Reply:
for address in previous_version...

In this part of code, we are iterating over the files in the previous slice to see, which have been deleted. If we find such a file, there is no "new" address to write in this slice, because the file was deleted. So we include these tokens into this slice (with "-" sign), but as an address we write an old address.